### PR TITLE
 refactored calculate_kpoint_conv() and calculate_encut_conv()

### PR DESCRIPTION
**Pull request by lukeamyers**
### refactored calculate_kpoint_conv() and calculate_encut_conv()

Relative paths are good for tutorials, but unfortunately that broke these two functions. Further because they errored that left the working directory in a changed state, which made it difficult to debug. So I was going to make a small change, but ended up significantly refactoring these functions. They still follow essentially the same steps.

- Unified the output df format for both, since they were so similar
image

- `extract_kpoints()` modified to return `list[int]` instead of `list[str]` (don't know if this will cause bugs elsewhere)
- both `calculate_kpoint_conv()` and `calculate_encut_conv()` are now basically convenience functions that call `extract_convergence_data()` and their respective plotting function (`plot_kpoint_conv()` or `plot_encut_conv()`).
- `extract_convergence_data()` is located in `aggregate_data_extraction.py`. I left everything else in `workflows.py`. I'm thinking maybe workflows should be only for the custodian type stuff maybe. not sure about that though.

### CODE REVIEW

**Comment by lukeamyers** [comment]
I forgot to add doc strings. let me do that

**Comment by nhew1994** [comment]
> ### refactored calculate_kpoint_conv() and calculate_encut_conv()
> Relative paths are good for tutorials, but unfortunately that broke these two functions. Further because they errored that left the working directory in a changed state, which made it difficult to debug. So I was going to make a small change, but ended up significantly refactoring these functions. They still follow essentially the same steps.
> 
> * Unified the output df format for both, since they were so similar
>   image
> * `extract_kpoints()` modified to return `list[int]` instead of `list[str]` (don't know if this will cause bugs elsewhere)
> * both `calculate_kpoint_conv()` and `calculate_encut_conv()` are now basically convenience functions that call `extract_convergence_data()` and their respective plotting function (`plot_kpoint_conv()` or `plot_encut_conv()`).
> * `extract_convergence_data()` is located in `aggregate_data_extraction.py`. I left everything else in `workflows.py`. I'm thinking maybe workflows should be only for the custodian type stuff maybe. not sure about that though.

You can move calculate_kpoint_conv(), calculate_encut_conv(), plot_kpoint_conv(), and plot_encut_conv()) to aggregate_extraction.py

File: dfttk/aggregate_extraction.py [hunk 1]
```diff
@@ -150,3 +156,194 @@ def recursive_extract_configuration_data(
             print(f"Error in {config_dir}: {e}")
     df = pd.concat(df_list, ignore_index=True)
     return df
+
+def extract_convergence_data(path: str) -> pd.DataFrame:
+    """Calculates the energy convergence with respect to ENCUT and plots the results.
```
Comment (nhew1994) [inline dfttk/aggregate_extraction.py:161]: Calculates the energy convergence with respect to ENCUT and kpoints and plots the results.

File: dfttk/aggregate_extraction.py [hunk 2]
```diff
@@ -150,3 +156,194 @@ def recursive_extract_configuration_data(
...
+
+    Args:
+        path (str): path to the folder containing the VASP input files
+        plot (bool, optional): If True, plots the energy per atom vs. ENCUT. Defaults to True.
+
+    Returns:
+        pd.DataFrame: a pandas dataframe containing the ENCUT, energy, number of atoms, energy per atom, and difference in energy per atom.
+        go.Figure: a plotly figure of the energy per atom vs. ENCUT.
```
Comment (nhew1994) [inline dfttk/aggregate_extraction.py:169]: pd.DataFrame: add kpoints columns

**Review by nhew1994 — APPROVED** [review]
I think just modify the docstring of extract_convergence_data to include kpoints and I am ok to merge.

**Pull request merged**